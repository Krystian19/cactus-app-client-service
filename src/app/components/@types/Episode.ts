import Anime from './Anime';
import Release from './Release';

export type Episode = {
  id: Number,
  thumbnail?: String,
  episodeOrder?: Number,
  EarlierEpisode?: Episode,
  LaterEpisode?: Episode,
  Release?: Release,
  Anime?: Anime,
};

export default Episode;
