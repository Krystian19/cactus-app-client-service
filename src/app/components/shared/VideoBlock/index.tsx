import React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link, withRouter } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import { injectIntl, InjectedIntlProps } from 'react-intl';
import { GQLEpisode } from '@cactus-app/types';

import LazyImage from '../LazyImage';

type PropType = RouteComponentProps<{}> & InjectedIntlProps & {
  episodes: GQLEpisode[];
  title: string;
  viewAllLink?: string;
};

const VideoBlock = (props: PropType): JSX.Element => {
  const {
    episodes,
    title,
    viewAllLink,
    history,
    intl: { formatMessage },
  } = props;

  return (
    <div className="video-block">
      <div className="video-block-title">
        <div className="video-block-title-head">
          <span>{title}</span>
        </div>
        {viewAllLink && (
          <Link to={String(viewAllLink)} className="button">
            <FormattedMessage
              id="cactus.view_all"
              defaultMessage={'View all'}
            />
          </Link>
        )}
      </div>
      <div className="video-block-content">
        <div className="anime-small-thumbnail-list">
          {episodes.map((episode, index) => (
            <div
              key={Number(episode.id)}
              className="anime-small-thumbnail"
              onClick={(): void => history.push(`/anime/video/${episode.id}`)}
              onKeyPress={(): void => history.push(`/anime/video/${episode.id}`)}
              role="menuitem"
              tabIndex={index}
            >
              <div className="cover shimmer-load">
                <LazyImage
                  src={`/img_cdn/${episode.Release.background}`}
                  errorSrc="/img/thumbnail_placeholder.png"
                  alt="thumbnail"
                  className="anime-small-thumbnail fade-in"
                  noLoadingPlaceholder={true}
                />
                <div className="overlay hover_hidden darken">
                  <svg className="play" viewBox="0 0 24 24">
                    <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
                  </svg>
                </div>
              </div>
              <div className="info">
                <div className="title">
                  <div className="title-container">
                    {`${episode.Release.title}
                    (${
                      formatMessage({
                        id: 'cactus.season_short',
                        defaultMessage: 'S'
                      })}${episode.Release.release_order})`}
                  </div>
                  <div className="detail-container">
                    {`Ep. ${episode.episode_order}`}
                  </div>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default withRouter(
  injectIntl(VideoBlock)
);
