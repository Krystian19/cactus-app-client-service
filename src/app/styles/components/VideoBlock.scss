.video-block {
  width: 80%;
  margin: 20px auto 50px;
}

.video-block .video-block-title {
  margin-bottom: 30px;
  display: flex;
}

.video-block .video-block-title .video-block-title-head {
  flex: 1;
}

.video-block .video-block-title .video-block-title-head span {
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1.28571429em;
  border-bottom: 2px solid #ff6c00;
  padding-bottom: .3em;
  margin-right: 20px;
}

.video-block .video-block-title .video-block-title-options {
  /* flex: 1; */
  display: flex;
  justify-content: flex-end;
}

.video-block .video-block-content {}