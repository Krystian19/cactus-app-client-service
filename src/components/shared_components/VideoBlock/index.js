import React from 'react';

// ({ Episodes }) => (
const VideoBlock = (props) => {
  const { props: { episodes } } = props;

  return (
    <div className="video-block">
      <div className="video-block-title">
        <div className="video-block-title-head">
          <span>{title}</span>
        </div>
        <div className="video-block-title-options">
          <a href="http://www.google.com" className="button">View all</a>
        </div>
      </div>
      <div className="video-block-content">
        <div className="anime-small-thumbnail-list">
          {episodes.map(episode => (
            <div key={episode.id} className="anime-small-thumbnail">
              <div className="cover">
                <img src={episode.thumbnail} alt="thumbnail" className="anime-small-thumbnail" />
                <div className="overlay hover_hidden darken">
                  <svg className="play" viewBox="0 0 24 24">
                    <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
                  </svg>
                </div>
              </div>
              <div className="info">
                <div className="title">
                  <div className="title-container">{`${episode.Season.title} (Season ${episode.Season.seasonOrder})`}</div>
                  <div className="detail-container">{`Ep. ${episode.episodeOrder}`}</div>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

// <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/2808dEyIDBR6.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> No Game No Life: Zero lorem </div>
//               <div className="detail-container"> Ep. 1 </div>
//             </div>
//           </div>
//         </div>
//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/28193ESeF49G.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Shokugeki no Souma 3rd season </div>
//               <div className="detail-container"> Ep. 12 </div>
//             </div>
//           </div>
//         </div>
//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/28174OVMKYWJ.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Tokyo Ghoul 3rd season </div>
//               <div className="detail-container"> Ep. 11 </div>
//             </div>
//           </div>
//         </div>
//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/2908UZUvpOMN.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Shingeki no Kyojin Season 3 </div>
//               <div className="detail-container"> Ep. 10 </div>
//             </div>
//           </div>
//         </div>

//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/1105x2klHygG.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Fate/stay night 2nd Season </div>
//               <div className="detail-container"> Ep. 13 </div>
//             </div>
//           </div>
//         </div>
//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/282mpLwPxe8.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Black Lagoon: 2nd season </div>
//               <div className="detail-container"> Ep. 10 </div>
//             </div>
//           </div>
//         </div>
//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/2823lknkr4jp.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Wotaku ni Koi wa Muzukashii </div>
//               <div className="detail-container"> Ep. 4 </div>
//             </div>
//           </div>
//         </div>
//         <div className="anime-small-thumbnail">
//           <div className="cover">
//             <img src="https://cdn.masterani.me/wallpaper/0/445H9PlfsZS.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//             <div className="overlay hover_hidden darken">
//               <svg className="play" viewBox="0 0 24 24">
//                 <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//               </svg>
//             </div>
//           </div>
//           <div className="info">
//             <div className="title">
//               <div className="title-container"> Yahari Ore no Seishun lorem </div>
//               <div className="detail-container"> Ep. 5 </div>
//             </div>
//           </div>
//         </div>


export default VideoBlock;
