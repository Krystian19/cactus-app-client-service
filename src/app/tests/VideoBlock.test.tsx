import React from 'react';
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider, addLocaleData } from 'react-intl';
import locale_en from 'react-intl/locale-data/en';
import locale_es from 'react-intl/locale-data/es';

import VideoBlock from '../components/shared/VideoBlock';
import Episode from '../components/@types/Episode';

const episodes: Array<Episode> = [
  {
    'id': 21,
    'thumbnail': 'thumbnail_placeholder.png',
    'episodeOrder': 12,
    'Release': {
      'id': 7,
      'releaseOrder': 1,
      'title': 'Goblin Slayer',
      'background': 'thumbnail_placeholder.png',
    },
  },
];

// Setup language locale support
addLocaleData([...locale_en, ...locale_es]);

const componentSetup = (
  <IntlProvider locale="en">
    <VideoBlock
      title={'Newest episodes'}
      episodes={episodes}
      viewAllLink={'/newest_episodes'}
    />
  </IntlProvider>
);

test('VideoBlock component is rendering properly', () => {
  const component = render(
    <MemoryRouter>
      {componentSetup}
    </MemoryRouter>
  );

  expect(component).toMatchSnapshot();
});
