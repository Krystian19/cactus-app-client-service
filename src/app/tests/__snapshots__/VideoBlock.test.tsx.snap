// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoBlock component is rendering properly 1`] = `
<div
  class="video-block"
>
  <div
    class="video-block-title"
  >
    <div
      class="video-block-title-head"
    >
      <span>
        Newest episodes
      </span>
    </div>
    <a
      class="button"
      href="/newest_episodes"
    >
      <span>
        View all
      </span>
    </a>
  </div>
  <div
    class="video-block-content"
  >
    <div
      class="anime-small-thumbnail-list"
    >
      <div
        class="anime-small-thumbnail"
        role="menuitem"
        tabindex="0"
      >
        <div
          class="cover shimmer-load"
        >
          <div
            class="overlay hover_hidden darken"
          >
            <svg
              class="play"
              viewBox="0 0 24 24"
            >
              <path
                d="M8,5.14V19.14L19,12.14L8,5.14Z"
              />
            </svg>
          </div>
        </div>
        <div
          class="info"
        >
          <div
            class="title"
          >
            <div
              class="title-container"
            >
              Goblin Slayer
                      (S1)
            </div>
            <div
              class="detail-container"
            >
              Ep. 12
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
